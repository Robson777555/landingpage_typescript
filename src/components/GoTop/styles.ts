import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;