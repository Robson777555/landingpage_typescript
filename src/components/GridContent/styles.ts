// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;