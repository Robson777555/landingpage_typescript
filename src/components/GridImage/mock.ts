interface GridItem {
  altText: string;
  srcImg: string;
}

interface MockData {
  title: string;
  description: string;
  grid: GridItem[];
  background?: boolean;
}

const mock: MockData = {
  background: false,
  title: '<b style="font-size: 2rem;">Tec</b>',
  description: 'Conhe√ßa as principais tecnologias usadas neste projeto:',
  grid: [
    {
      altText: 'JavaScript Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/6/6a/JavaScript-logo.png',
    },
    {
      altText: 'React Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg',
    },
    {
      altText: 'Next.js Logo',
      srcImg: 'https://assets.vercel.com/image/upload/v1662090959/front/nextjs/twitter-card.png',
    },
    {
      altText: 'Node.js Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Node.js_logo.svg',
    },
    {
      altText: 'Git Logo',
      srcImg: 'https://git-scm.com/images/logos/downloads/Git-Icon-1788C.png',
    },
  ],
};

export default mock;