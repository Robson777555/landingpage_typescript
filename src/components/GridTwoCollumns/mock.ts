interface MockData {
  title: string;
  text: string;
  srcImg: string;
  background?: boolean;
}

const mock: MockData = {
  title: 'Essenciais',
  text: 'As 3 tecnologias básicas no front são:',
  srcImg: '/assets/images/javascript.svg',
  background: false,
};

export default mock;