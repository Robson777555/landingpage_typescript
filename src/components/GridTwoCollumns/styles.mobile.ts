import styled, { css } from 'styled-components';

export const MobileGridTwoCol = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    width: 100%;
    max-width: 98vw;
    min-width: 0;
    padding: ${theme.spacings.small};
    box-sizing: border-box;

    h2, h3, h4, h5, h6 {
      font-size: 2rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }
    p, span, div {
      font-size: 1rem;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.3;
    }
    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }
  `}
`;
