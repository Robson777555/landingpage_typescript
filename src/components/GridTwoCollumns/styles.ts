// src/components/GridTwoCollumns/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      text-align: center;
      justify-content: center;
      gap: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.medium};

    /* Estilos para mobile */
    @media ${theme.media.lteMedium} {
      align-items: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.medium};
    max-width: 400px;
    width: 100%;

    img, .next-image {
      max-width: 100%;
      height: auto;
      width: 100%;
      object-fit: contain;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} 0;
      max-width: 300px;
    }
  `}
`;

export const Title = styled.h2`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
