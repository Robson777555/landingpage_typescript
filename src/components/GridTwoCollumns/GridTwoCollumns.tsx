"use client";
import React from "react";
import Image from "next/image";
import * as Styled from "./styles";
import { SectionBackground } from "../SectionBackground/SectionBackground";
import { TextComponent } from "../TextComponent/TextComponent";
import { MobileGridTwoCol } from "./styles.mobile";
import { useIsMobile } from "../../hooks/useIsMobile";

interface GridTwoCollumnsProps {
  title: string;
  text: string;
  srcImg: string;
  background?: boolean;
}

export const GridTwoCollumns = ({
  title,
  text,
  srcImg,
  background = false,
}: GridTwoCollumnsProps) => {
  const isMobile = useIsMobile();

  if (isMobile) {
    return (
      <MobileGridTwoCol>
        <h2 style={{ fontWeight: "bold" }}>{title}</h2>
        <p>{text}</p>
        <img src={srcImg} alt={title} />
      </MobileGridTwoCol>
    );
  }

  return (
    <SectionBackground background={background}>
      <Styled.Container>
        <Styled.TextContainer>
          <Styled.Title>{title}</Styled.Title>
          <TextComponent>{text}</TextComponent>
        </Styled.TextContainer>
        <Styled.ImageContainer>
          <Image
            src={srcImg}
            alt={title}
            width={500}
            height={500}
            style={{
              width: "100%",
              height: "auto",
              maxWidth: "400px",
            }}
          />
        </Styled.ImageContainer>
      </Styled.Container>
    </SectionBackground>
  );
};
