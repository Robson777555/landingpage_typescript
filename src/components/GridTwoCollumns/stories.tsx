
import { GridTwoCollumns } from './GridTwoCollumns';
import mock from './mock';
import { Meta, StoryObj } from '@storybook/react';

export default {
  title: 'GridTwoCollumns',
  component: GridTwoCollumns,
  args: mock,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta<typeof GridTwoCollumns>;

type Story = StoryObj<typeof GridTwoCollumns>;

export const Default: Story = {};

export const WithBackground: Story = {
  args: {
    ...mock,
    background: true,
  },
};