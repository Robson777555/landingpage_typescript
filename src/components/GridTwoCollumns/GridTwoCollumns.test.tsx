import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { GridTwoCollumns } from './GridTwoCollumns';
import { describe, it, expect, vi, Mock } from 'vitest';
import mock from './mock';
import { useIsMobile } from '../../hooks/useIsMobile';

// Mock do hook useIsMobile
vi.mock('../../hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}));

describe('<GridTwoCollumns />', () => {
  it('should render title, text and image for desktop version', () => {
    // Configura o mock para desktop
    (useIsMobile as Mock).mockReturnValue(false);

    render(
      <ThemeProvider theme={theme}>
        <GridTwoCollumns {...mock} />
      </ThemeProvider>
    );

    // Verifica os elementos
    expect(screen.getByRole('heading', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(mock.text)).toBeInTheDocument();

    // Verifica a imagem (componente Image do Next.js)
    const image = screen.getByRole('img', { name: mock.title });
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', mock.srcImg);
  });

  it('should render mobile version when isMobile is true', () => {
    // Configura o mock para mobile
    (useIsMobile as Mock).mockReturnValue(true);

    render(
      <ThemeProvider theme={theme}>
        <GridTwoCollumns {...mock} />
      </ThemeProvider>
    );

    // Verifica os elementos
    expect(screen.getByRole('heading', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(mock.text)).toBeInTheDocument();

    // Verifica a imagem (tag img normal no mobile)
    const image = screen.getByRole('img', { name: mock.title });
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', mock.srcImg);
  });

  it('should match snapshot for desktop version', () => {
    (useIsMobile as Mock).mockReturnValue(false);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridTwoCollumns {...mock} />
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot for mobile version', () => {
    (useIsMobile as Mock).mockReturnValue(true);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridTwoCollumns {...mock} />
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with background when background prop is true', () => {
    (useIsMobile as Mock).mockReturnValue(false);

    render(
      <ThemeProvider theme={theme}>
        <GridTwoCollumns {...mock} background={true} />
      </ThemeProvider>
    );

    // Verifica se os elementos estÃ£o renderizados com fundo
    expect(screen.getByRole('heading', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(mock.text)).toBeInTheDocument();
    expect(screen.getByRole('img', { name: mock.title })).toBeInTheDocument();
  });
  it ('Snapshot of page', () => {
    (useIsMobile as Mock).mockReturnValue(false);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridTwoCollumns {...mock} />
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  })
});