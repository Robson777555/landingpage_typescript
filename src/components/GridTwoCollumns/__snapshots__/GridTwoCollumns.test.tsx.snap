// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridTwoCollumns /> > Snapshot of page 1`] = `
<div
  class="sc-dYOLZc fAiKqp"
  data-testid="section-background"
>
  <div
    class="sc-jMpmlX CjQRF"
    data-testid="section-container"
  >
    <div
      class="sc-bXdtCk jFQVhJ"
    >
      <div
        class="sc-hKDTPf bzHkyl"
      >
        <h2
          class="sc-gwsNht mBpBH"
        >
          Essenciais
        </h2>
        <div
          class="sc-hoLldG fOqDwp"
          data-testid="text-component"
        >
          As 3 tecnologias básicas no front são:
        </div>
      </div>
      <div
        class="sc-fmZSGO bDmhIR"
      >
        <img
          alt="Essenciais"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/assets/images/javascript.svg"
          style="color: transparent; width: 100%; height: auto; max-width: 400px;"
          width="500"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridTwoCollumns /> > should match snapshot for desktop version 1`] = `
<div
  class="sc-dYOLZc fAiKqp"
  data-testid="section-background"
>
  <div
    class="sc-jMpmlX CjQRF"
    data-testid="section-container"
  >
    <div
      class="sc-bXdtCk jFQVhJ"
    >
      <div
        class="sc-hKDTPf bzHkyl"
      >
        <h2
          class="sc-gwsNht mBpBH"
        >
          Essenciais
        </h2>
        <div
          class="sc-hoLldG fOqDwp"
          data-testid="text-component"
        >
          As 3 tecnologias básicas no front são:
        </div>
      </div>
      <div
        class="sc-fmZSGO bDmhIR"
      >
        <img
          alt="Essenciais"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/assets/images/javascript.svg"
          style="color: transparent; width: 100%; height: auto; max-width: 400px;"
          width="500"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridTwoCollumns /> > should match snapshot for mobile version 1`] = `
<div
  class="sc-jZhnRx kNAgFn"
>
  <h2
    style="font-weight: bold;"
  >
    Essenciais
  </h2>
  <p>
    As 3 tecnologias básicas no front são:
  </p>
  <img
    alt="Essenciais"
    src="/assets/images/javascript.svg"
  />
</div>
`;
